api-handler/index.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+const Boom = require('@hapi/boom');
+const generateSitemapFromEntries = require('./generate');
+const handleDynamicRoutesMapping = require('./mapping');
+
+const handleErrorDefault = (err) => {
+    // Only log internal server errors
+    if (!err.isServer) {
+        return;
+    }
+
+    // Log original error if passed
+    if (err.data && err.data.originalError) {
+        err = err.data.originalError;
+    }
+
+    console.error(err.stack);
+};
+
+const handleWarningDefault = (message) => console.warn(message);
+
+const sendError = (res, err) => {
+    const { output } = err;
+    const { headers, statusCode, payload } = output;
+
+    Object.entries(headers).forEach(([key, value]) => res.setHeader(key, value));
+
+    res.status(statusCode).json(payload);
+};
+
+const createSitemap = async ({ baseUrl, mapDynamicRoutes, handleWarning }) => {
+    const sitemapEntries = global.__NEXT_ROUTES__;
+
+    if (!sitemapEntries) {
+        throw new Error('There are no entries to map. You might want to check the __NEXT_ROUTES__ global variable.');
+    }
+
+    const mappedEntries = await handleDynamicRoutesMapping(sitemapEntries, { handleWarning, mapDynamicRoutes });
+    const sitemapXml = generateSitemapFromEntries(mappedEntries, { baseUrl });
+
+    return sitemapXml;
+};
+
+/**
+ * API handler to generate a mapped sitemap.xml.
+ *
+ * @param {object} options - The options.
+ * @param {string} options.baseUrl - The website url to join on each entry.
```

Comment:
Please update all the JSDocs descriptions if you accept my changes on the README so that they match.